c7n/resources/iam.py

```diff
@@ -774,6 +774,120 @@ def _user_mfa_devices(resource):
         return matched
 
 
+@User.filter_registry.register('username')
+class UserName(ValueFilter):
```

Comment:
unclear this is needed, the default value filter on username can address subattributes directly, the key in a value filters is a jmepath expression that allows for this directly.